### ISSUE TITLE
Visual F# Colorization, Intellisense super slow

### ISSUE THREAD

**Issue by realvictorprm** [Microsoft/visualfsharp#3054]
The Visual F# extension is creppy slow on my laptop and for @NatElkins since Friday last week.
Meant is that both colorization and intellisense take ages for showing and updating content so that nice coding isn't possible.

#### Repro steps

1. Open a solution which contains F# projects
2. Open a F# file
3. Wait...

#### Expected behavior

Fast colorization and Intellisense.

#### Actual behavior

Colorization takes more than one minute to appear, even longer.
Intellisense needs more than half an minute to appear after the Colorization appeared.

#### Known workarounds

*Downgrade*

#### Related information 

Provide any related information 

* Operating system: Win10
* Branch: Latest nightlies so master

**Comment by dsyme** [comment]
@realvictorprm 
- A repro would be great please.  
- I'm wondering if this is related to the fix in #3002. This may have enabled "cross- project" references in more situations, 
- Does this happen on every time you change a file or only on the first colorization?

**Comment by NatElkins** [comment]
I notice that in VisualFSharp.sln if I go to il.fs I get colorization and Intellisense quickly and without issue.  When I go to something like MSBuildUtilities, it takes much, much longer.

**Comment by NatElkins** [comment]
Another piece of anecdotal evidence: the "lower down" in the stack my projects are (i.e. less transitive dependencies, things like domain types) the faster I get colorization and Intellisense.

**Comment by dsyme** [comment]
> Another piece of anecdotal evidence: the "lower down" in the stack my projects are (i.e. less transitive dependencies, things like domain types) the faster I get colorization and Intellisense.

Asking agani: is that first-time colorization (open a file), or iterative colorization (after changing a file)? thanks

**Comment by realvictorprm** [comment]
@dsyme I'm very sorry that I cannot provide that much information on this.

first-time colorization

**Comment by dsyme** [comment]
@realvictorprm Thanks!

**Comment by dsyme** [comment]
@realvictorprm @NatElkins It is possible we will turn off cross-project checking (without compilation - i.e. checking against in-memory files rather than compiled DLLs on disk) by default and make it opt-in.

**Comment by realvictorprm** [comment]
I'm not yet into the details @dsyme. I would debug the stuff but debugging isn't even possible on my main computer. Too slow.

**Comment by 0x53A** [comment]
I sometimes have the same issue - colorization takes minutes or it is seemingly never colorized - is there something I can do to help?

What I could do is attach a debugger while it is uncolored and check what it is doing at that moment - any pointers where to look for that?

I do not have a reliable reproduction yet ...

**Comment by dsyme** [comment]
@0x53A You may find your problems are less severe on master after #3061 

@realvictorprm  @0x53A Re this:

>I sometimes have the same issue - colorization takes minutes or it is seemingly never colorized (that is, first time colorization after opening the file) - is there something I can do to help?

Try this:

1. get and build master
1. Start administrator VisualFSharp.sln 
1. Choose startup project VisualFSharpOpenSource
1. Choose "Release" configuration
1. Hit F5 to build and launch VS experimental (lets call that your "target VS")
1. In your target VS load your project and get ready to do your stuff
1. In your debugger clear the Output window
1. In your target VS do your stuff
1. In your debugger: grab the Output window and paste it into a gist
1. exit

We only want the accurate part of the log corresponding to the time period over which colorization was not active.

BTW is it all kinds of colorization or just semantic colorization?

**Comment by vasily-kirichenko** [comment]
I can just say that I never ever seen "no colorization" bug (I've not updated the main VS instance for a week however).

**Comment by dsyme** [comment]
@vasily-kirichenko I've not seen a complete lack of colorization for some time either.  (Long delays before semantic colorization appears in large multi-project solutions can be easily explained by the issue described in https://github.com/Microsoft/visualfsharp/pull/3063)

**Comment by 0x53A** [comment]
@dsyme I currently have a case where one file seemingly gets never colorised (I waited a few minutes), where other files will get colorised after a short time.

The interesting thing is that there is NO output when opening this file, but I can see some chatter when opening other .fs files.

I noticed that a first-chance COMException get's thrown each time I open this file, this does not happen in the good case.

![image](https://cloud.githubusercontent.com/assets/4236651/26155174/9b465c0e-3b12-11e7-9735-de52d0ee4e82.png)


Attached is the full callstack

```

>	Microsoft.VisualStudio.Design.dll!Microsoft.VisualStudio.Design.VSDesignSurfaceManager.Microsoft.VisualStudio.Shell.Interop.IVsSelectionEvents.OnElementValueChanged(uint elementid, object varValueOld, object varValueNew)	Unknown
 	[Native to Managed Transition]	
 	msenv.dll!CCookieTable<struct std::pair<unsigned long,class ATL::CAdapt<class ATL::CComPtr<struct IVsSelectionEvents> > >,struct DefaultCookieTraits<unsigned long,1,4294967295,0,1>,struct DefaultValueTraits<struct std::pair<unsigned long,class ATL::CAdapt<class ATL::CComPtr<struct IVsSelectionEvents> > > > >::for_each<class FireOnElementValueChanged>(class FireOnElementValueChanged,bool)	Unknown
 	msenv.dll!CVsMonSel::NotifySelElemChangeInternal(unsigned long,struct tagSELELEM &)	Unknown
 	msenv.dll!CVsMonSel::NotifyChangedElemsHelper(class std::vector<struct std::pair<bool,struct tagSELELEM>,class std::allocator<struct std::pair<bool,struct tagSELELEM> > > &,int)	Unknown
 	msenv.dll!CVsMonSel::RegisterSelCtx(struct IVsTrackSelectionEx *)	Unknown
 	msenv.dll!CVsSelCtx::Register(void)	Unknown
 	Microsoft.VisualStudio.Platform.WindowManagement.ni.dll!62fed9c7()	Unknown
 	[Frames below may be incorrect and/or missing, native debugger attempting to walk managed call stack]	
 	[Managed to Native Transition]	
 	Microsoft.VisualStudio.Platform.WindowManagement.dll!Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.ConnectSelectionContext()	Unknown
 	Microsoft.VisualStudio.Platform.WindowManagement.dll!Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.Activate()	Unknown
 	Microsoft.VisualStudio.Platform.WindowManagement.dll!Microsoft.VisualStudio.Platform.WindowManagement.WindowManagerService.viewManager_ActiveViewChanged(object sender, Microsoft.VisualStudio.PlatformUI.Shell.ActiveViewChangedEventArgs e)	Unknown
 	Microsoft.VisualStudio.Utilities.dll!Microsoft.VisualStudio.PlatformUI.ExtensionMethods.RaiseEvent<System.__Canon>(System.EventHandler<System.__Canon> eventHandler, object source, System.__Canon args)	Unknown
 	Microsoft.VisualStudio.Shell.ViewManager.dll!Microsoft.VisualStudio.PlatformUI.Shell.ViewManager.SetActiveView(Microsoft.VisualStudio.PlatformUI.Shell.View view, Microsoft.VisualStudio.PlatformUI.Shell.ActivationType type)	Unknown
 	Microsoft.VisualStudio.Platform.WindowManagement.dll!Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.ShowInternal(Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.ShowFlags showFlags)	Unknown
 	Microsoft.VisualStudio.Platform.WindowManagement.dll!Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.Show.AnonymousMethod__504_0()	Unknown
 	Microsoft.VisualStudio.Shell.15.0.dll!Microsoft.VisualStudio.ErrorHandler.CallWithCOMConvention(System.Func<int> method, bool reportError = false, bool setShellErrorInfo = true)	Unknown
 	Microsoft.VisualStudio.Platform.WindowManagement.dll!Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.Show()	Unknown
 	Microsoft.VisualStudio.Platform.WindowManagement.dll!Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.MarshalingWindowFrame.Microsoft.VisualStudio.Shell.Interop.IVsWindowFrame.Show.AnonymousMethod__12_0()	Unknown
 	Microsoft.VisualStudio.Shell.15.0.dll!Microsoft.VisualStudio.Shell.ThreadHelper.Invoke<int>(System.Func<int> method)	Unknown
 	Microsoft.VisualStudio.Platform.WindowManagement.dll!Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.MarshalingWindowFrame.Microsoft.VisualStudio.Shell.Interop.IVsWindowFrame.Show()	Unknown
 	FSharp.ProjectSystem.Base.dll!Microsoft.VisualStudio.FSharp.ProjectSystem.FileDocumentManager.Open(bool newFile, bool openWith, uint editorFlags, ref System.Guid editorType, string physicalView, ref System.Guid logicalView, System.IntPtr docDataExisting, out Microsoft.VisualStudio.Shell.Interop.IVsWindowFrame windowFrame = {Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.MarshalingWindowFrame}, Microsoft.VisualStudio.FSharp.ProjectSystem.WindowFrameShowAction windowFrameAction)	Unknown
 	FSharp.ProjectSystem.Base.dll!Microsoft.VisualStudio.FSharp.ProjectSystem.FileDocumentManager.Open(bool newFile, bool openWith, ref System.Guid logicalView, System.IntPtr docDataExisting, out Microsoft.VisualStudio.Shell.Interop.IVsWindowFrame windowFrame, Microsoft.VisualStudio.FSharp.ProjectSystem.WindowFrameShowAction windowFrameAction)	Unknown
 	FSharp.ProjectSystem.Base.dll!Microsoft.VisualStudio.FSharp.ProjectSystem.FileDocumentManager.Open(bool newFile, bool openWith, System.Guid logicalView, out Microsoft.VisualStudio.Shell.Interop.IVsWindowFrame frame, Microsoft.VisualStudio.FSharp.ProjectSystem.WindowFrameShowAction windowFrameAction)	Unknown
 	FSharp.ProjectSystem.FSharp.dll!Microsoft.VisualStudio.FSharp.ProjectSystem.FSharpFileNode.DoDefaultAction()	Unknown
 	FSharp.ProjectSystem.Base.dll!Microsoft.VisualStudio.FSharp.ProjectSystem.HierarchyNode.ExecCommandOnNode(System.Guid cmdGroup, uint cmd, uint nCmdexecopt, System.IntPtr pvaIn, System.IntPtr pvaOut)	Unknown
 	FSharp.ProjectSystem.Base.dll!Microsoft.VisualStudio.FSharp.ProjectSystem.FileNode.ExecCommandOnNode(System.Guid cmdGroup, uint cmd, uint nCmdexecopt, System.IntPtr pvaIn, System.IntPtr pvaOut)	Unknown
 	FSharp.ProjectSystem.FSharp.dll!Microsoft.VisualStudio.FSharp.ProjectSystem.FSharpFileNode.ExecCommandOnNode(System.Guid guidCmdGroup, uint cmd, uint nCmdexecopt, System.IntPtr pvaIn, System.IntPtr pvaOut)	Unknown
 	FSharp.ProjectSystem.Base.dll!Microsoft.VisualStudio.FSharp.ProjectSystem.HierarchyNode.InternalExecCommand(System.Guid cmdGroup, uint cmdId = 0x00000002, uint cmdExecOpt = 0x00000000, System.IntPtr vaIn = 0x00000000, System.IntPtr vaOut = 0x00000000, Microsoft.VisualStudio.FSharp.ProjectSystem.CommandOrigin commandOrigin)	Unknown
 	FSharp.ProjectSystem.Base.dll!Microsoft.VisualStudio.FSharp.ProjectSystem.HierarchyNode.ExecCommand(uint itemId, ref System.Guid guidCmdGroup, uint nCmdId, uint nCmdExecOpt, System.IntPtr pvain, System.IntPtr p)	Unknown
 	[Native to Managed Transition]	
 	ProjectAggregator.dll!CProjectAggregator::ExecCommand(unsigned long,struct _GUID const *,unsigned long,unsigned long,struct tagVARIANT *,struct tagVARIANT *)	Unknown
 	Microsoft.VisualStudio.Shell.Interop.ni.dll!66720174()	Unknown
 	[Managed to Native Transition]	
 	Microsoft.VisualStudio.Shell.15.0.dll!Microsoft.Internal.VisualStudio.PlatformUI.HierarchyUtilities.ExecHierParentChain.AnonymousMethod__1()	Unknown
 	Microsoft.VisualStudio.Shell.15.0.dll!Microsoft.VisualStudio.ErrorHandler.CallWithCOMConvention(System.Func<int> method, bool reportError = false, bool setShellErrorInfo = true)	Unknown
 	Microsoft.VisualStudio.Shell.15.0.dll!Microsoft.Internal.VisualStudio.PlatformUI.HierarchyUtilities.ExecHierParentChain(Microsoft.VisualStudio.Shell.IVsHierarchyItemManager manager, Microsoft.VisualStudio.Shell.Interop.IVsUIHierarchy lpUIHCmd, Microsoft.VisualStudio.Shell.Interop.IVsUIHierarchy lpUIHCurrent, uint itemidCurrent, ref System.Guid pguidCmdGroupRef, uint nCmdID, uint nCmdexecopt, System.IntPtr pvaIn, System.IntPtr pvaOut)	Unknown
 	Microsoft.VisualStudio.Shell.TreeNavigation.HierarchyProvider.dll!Microsoft.VisualStudio.PlatformUI.HierarchyItem.HierarchyInvocationController.Invoke(System.Collections.Generic.IEnumerable<object> items, Microsoft.Internal.VisualStudio.PlatformUI.InputSource inputSource, bool preview)	Unknown
 	Microsoft.VisualStudio.Shell.15.0.dll!Microsoft.Internal.VisualStudio.PlatformUI.InvocationController.Invoke(System.Collections.Generic.IEnumerable<Microsoft.Internal.VisualStudio.PlatformUI.IInvocationPattern> items, Microsoft.Internal.VisualStudio.PlatformUI.InputSource inputSource = Mouse, bool preview = false, System.Func<Microsoft.Internal.VisualStudio.PlatformUI.IInvocationPattern, Microsoft.Internal.VisualStudio.PlatformUI.IInvocationController> getController)	Unknown
 	Microsoft.VisualStudio.Shell.15.0.dll!Microsoft.Internal.VisualStudio.PlatformUI.InvocationController.Invoke(System.Collections.Generic.IEnumerable<Microsoft.Internal.VisualStudio.PlatformUI.IInvocationPattern> items, Microsoft.Internal.VisualStudio.PlatformUI.InputSource inputSource, bool preview)	Unknown
 	Microsoft.VisualStudio.Shell.15.0.dll!Microsoft.Internal.VisualStudio.PlatformUI.PivotTreeViewItem.InvokeItem(System.Windows.Input.InputDevice sourceDevice)	Unknown
 	Microsoft.VisualStudio.Shell.15.0.dll!Microsoft.Internal.VisualStudio.PlatformUI.PivotTreeViewItem.OnMouseLeftButtonDown(System.Windows.Input.MouseButtonEventArgs e)	Unknown
 	PresentationCore.dll!System.Windows.UIElement.OnMouseLeftButtonDownThunk(object sender, System.Windows.Input.MouseButtonEventArgs e)	Unknown
 	PresentationCore.dll!System.Windows.Input.MouseButtonEventArgs.InvokeEventHandler(System.Delegate genericHandler, object genericTarget)	Unknown
 	PresentationCore.dll!System.Windows.RoutedEventArgs.InvokeHandler(System.Delegate handler, object target)	Unknown
 	PresentationCore.dll!System.Windows.RoutedEventHandlerInfo.InvokeHandler(object target, System.Windows.RoutedEventArgs routedEventArgs)	Unknown
 	PresentationCore.dll!System.Windows.EventRoute.InvokeHandlersImpl(object source = {Microsoft.Internal.VisualStudio.PlatformUI.PivotTreeViewItem}, System.Windows.RoutedEventArgs args = {System.Windows.Input.MouseButtonEventArgs}, bool reRaised = true)	Unknown
 	PresentationCore.dll!System.Windows.UIElement.ReRaiseEventAs(System.Windows.DependencyObject sender = {Microsoft.Internal.VisualStudio.PlatformUI.PivotTreeViewItem}, System.Windows.RoutedEventArgs args = {System.Windows.Input.MouseButtonEventArgs}, System.Windows.RoutedEvent newEvent)	Unknown
 	PresentationCore.dll!System.Windows.UIElement.OnMouseDownThunk(object sender, System.Windows.Input.MouseButtonEventArgs e)	Unknown
 	PresentationCore.dll!System.Windows.Input.MouseButtonEventArgs.InvokeEventHandler(System.Delegate genericHandler, object genericTarget)	Unknown
 	PresentationCore.dll!System.Windows.RoutedEventArgs.InvokeHandler(System.Delegate handler, object target)	Unknown
 	PresentationCore.dll!System.Windows.RoutedEventHandlerInfo.InvokeHandler(object target, System.Windows.RoutedEventArgs routedEventArgs)	Unknown
 	PresentationCore.dll!System.Windows.EventRoute.InvokeHandlersImpl(object source = {System.Windows.Documents.Run}, System.Windows.RoutedEventArgs args = {System.Windows.Input.MouseButtonEventArgs}, bool reRaised = false)	Unknown
 	PresentationCore.dll!System.Windows.UIElement.RaiseEventImpl(System.Windows.DependencyObject sender = {System.Windows.Documents.Run}, System.Windows.RoutedEventArgs args = {System.Windows.Input.MouseButtonEventArgs})	Unknown
 	PresentationCore.dll!System.Windows.ContentElement.RaiseTrustedEvent(System.Windows.RoutedEventArgs args = {System.Windows.Input.MouseButtonEventArgs})	Unknown
 	PresentationCore.dll!System.Windows.Input.InputManager.ProcessStagingArea()	Unknown
 	PresentationCore.dll!System.Windows.Input.InputManager.ProcessInput(System.Windows.Input.InputEventArgs input)	Unknown
 	PresentationCore.dll!System.Windows.Input.InputProviderSite.ReportInput(System.Windows.Input.InputReport inputReport)	Unknown
 	PresentationCore.dll!System.Windows.Interop.HwndMouseInputProvider.ReportInput(System.IntPtr hwnd, System.Windows.Input.InputMode mode, int timestamp, System.Windows.Input.RawMouseActions actions, int x, int y, int wheel)	Unknown
 	PresentationCore.dll!System.Windows.Interop.HwndMouseInputProvider.FilterMessage(System.IntPtr hwnd = 0x00131378, MS.Internal.Interop.WindowMessage msg = WM_LBUTTONDOWN, System.IntPtr wParam = 0x00000001, System.IntPtr lParam = 0x01700602, ref bool handled = false)	Unknown
 	PresentationCore.dll!System.Windows.Interop.HwndSource.InputFilterMessage(System.IntPtr hwnd = 0x00131378, int msg = 0x00000201, System.IntPtr wParam = 0x00000001, System.IntPtr lParam = 0x01700602, ref bool handled = false)	Unknown
 	WindowsBase.dll!MS.Win32.HwndWrapper.WndProc(System.IntPtr hwnd = 0x00131378, int msg = 0x00000201, System.IntPtr wParam = 0x00000001, System.IntPtr lParam = 0x01700602, ref bool handled = false)	Unknown
 	WindowsBase.dll!MS.Win32.HwndSubclass.DispatcherCallbackOperation(object o)	Unknown
 	WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate callback, object args, int numArgs)	Unknown
 	WindowsBase.dll!System.Windows.Threading.ExceptionWrapper.TryCatchWhen(object source = {System.Windows.Threading.Dispatcher}, System.Delegate callback, object args, int numArgs, System.Delegate catchHandler = null)	Unknown
 	WindowsBase.dll!System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority priority, System.TimeSpan timeout, System.Delegate method, object args, int numArgs)	Unknown
 	WindowsBase.dll!MS.Win32.HwndSubclass.SubclassWndProc(System.IntPtr hwnd = 0x00131378, int msg = 0x00000201, System.IntPtr wParam = 0x00000001, System.IntPtr lParam = 0x01700602)	Unknown
 	[Native to Managed Transition]	
 	user32.dll!__InternalCallWinProc@20()	Unknown
 	user32.dll!UserCallWinProcCheckWow()	Unknown
 	user32.dll!DispatchMessageWorker()	Unknown
 	user32.dll!_DispatchMessageW@4()	Unknown
 	msenv.dll!MainMessageLoop::ProcessMessage(struct IMsoStdComponentMgr *,struct IVsWindowManager *,struct tagMSG &)	Unknown
 	msenv.dll!CMsoCMHandler::EnvironmentMsgLoop(void)	Unknown
 	msenv.dll!CMsoCMHandler::FPushMessageLoop(unsigned long)	Unknown
 	msenv.dll!SCM::FPushMessageLoop(class SCMI *,unsigned long,void *)	Unknown
 	msenv.dll!SCM_MsoCompMgr::FPushMessageLoop(unsigned long,unsigned long,void *)	Unknown
 	msenv.dll!CMsoComponent::PushMsgLoop(unsigned long)	Unknown
 	msenv.dll!VStudioMainLogged(void)	Unknown
 	msenv.dll!_VStudioMain()	Unknown
 	devenv.exe!000fdcc0()	Unknown
 	devenv.exe!000feafe()	Unknown
 	devenv.exe!0010eb96()	Unknown
 	devenv.exe!0010e6a9()	Unknown
 	kernel32.dll!778d62c4()	Unknown
 	ntdll.dll!__RtlUserThreadStart()	Unknown
 	ntdll.dll!__RtlUserThreadStart@8()	Unknown

```

This is with the current nightly, which should (I didn't check) already contain your changes:

![image](https://cloud.githubusercontent.com/assets/4236651/26155247/d622216e-3b12-11e7-825d-5e0b8989a5f3.png)


![image](https://cloud.githubusercontent.com/assets/4236651/26155293/fc874bd6-3b12-11e7-8977-d79c79dbf0e5.png)

**Comment by dsyme** [comment]
@0x53A Can you share a step-by-step repro for that please? If the code is open... Thanks

**Comment by 0x53A** [comment]
> If the code is open

Nope, sorry :\
This specific solution is closed source, 60 C#, 5 F#, 1 C++/CLI.

But I doesn't happen reliably, and I think (but I may mis-remember) I have seen the same some time ago in a newly created blank F# project.

What I can do is debug it if you can tell me where to look.

**Comment by 0x53A** [comment]
A few more observations from the last few days:

* I have one project I can more or less reliably trigger the no-colorization-at-all.
* I tried extracting that project from it's solution, but then the bug stopped occuring, so I have nothing I can share at the moment. :\
* If I clean the project and restart VS, the project gets colored OK.
* If I build the project and restart VS, it will be uncolored again.

**Comment by 0x53A** [comment]
This issue happens reliably in a base-lib which depends on 5 other projects. If I remove everything except these six projects from the solution, then the error stops happening, even though there should be no change from the view of these projects.

For me it looks like something completely unrelated in the solution fucks VS up.

The whole solution has ~60 projects, so I'm not going to try all permutations and check which of these causes this.

For now I'm dropping VS and using Rider.

**Comment by vasily-kirichenko** [comment]
> For now I'm dropping VS and using Rider.

👍

**Comment by dsyme** [comment]
Guys, don't troll please :)

**Comment by realvictorprm** [comment]
Thanks, I agree on that. I'll need to look into this again too...

Am 19.06.2017 21:39 schrieb "Don Syme" <notifications@github.com>:

**Comment by vasily-kirichenko** [comment]
I was not trolling.

**Comment by dsyme** [comment]
@vasily-kirichenko OK

**Comment by dsyme** [comment]
@realvictorprm I believe much of this is addressed by #3238. Could you try that out (or nightly once integrated)?  

@0x53A It is unclear to me if #3238 addresses your issues, which seem about correctness not perf

There are undoubtedly some remaining sources of slowness. Please resubmit a new issue with repro steps for any specific cases. thanks

**Comment by 0x53A** [comment]
@dsyme yeah, my issue is not time based. I can try out the next master (later ...) anyway, but I don't have any hope that that will magically fix it.

**Comment by cartermp** [comment]
@0x53A with #3061 merged, can you try out the latest master to see if it helps?

**Comment by dsyme** [comment]
@0x53A Also try master with #3328 merged and deselect the option

**Comment by 0x53A** [comment]
The latest nightly is from tuesday:

![image](https://user-images.githubusercontent.com/4236651/28206222-5b3f1c0e-6885-11e7-8e7c-531adec464ca.png)

-------------------------

@dsyme  @cartermp 

with a locally built master:

![image](https://user-images.githubusercontent.com/4236651/28207258-4e1a3d3e-6889-11e7-94ad-d19b4f2f3ec9.png)


This is with the latest 15.2, do I need to test it with 15.3 preview X?

**Comment by dsyme** [comment]
That property page is an unfortunate legacy of how things were done in 15.0/15.1.  See the screen shot here ![image](https://user-images.githubusercontent.com/7204669/28167069-2c3f9b80-67d2-11e7-8749-c31318328d18.png)

**Comment by dsyme** [comment]
Text Editor --> F# --> IntelliSense 

But the PR only just got merged, so it's not in the nightly yet

**Comment by 0x53A** [comment]
ah, so I looked in the wrong page.

I still can't find the new setting:
![image](https://user-images.githubusercontent.com/4236651/28210558-43e6533e-6898-11e7-9d81-8a1f8162607c.png)

Could this be related to the roslyn hive? Do I need to clobber the main instance? Or did I just mess up something during build / deploy?

Thanks!

**Comment by cartermp** [comment]
Closing this, as @0x53A's issue is in #3222 and there is now an option to disable in-memory cross-project references.

In the future, the in-memory references should have a good heuristic in place so that this option is rarely used.  But that's a different discussion to have.
